coq/server/registrants/autocmds.py

```diff
@@ -126,6 +130,61 @@ async def cont() -> None:
         go(nvim, aw=cont())
 
 
+
+
+def _load_user_defined_snips(nvim: Nvim, stack: Stack) -> None:
+    ultisnips_paths = []
+    neosnippet_paths = []
+    for path_str in nvim.list_runtime_paths():
+        path = Path(path_str)
+        if (neosnippet_path := path / "neosnippets").exists():
+            neosnippet_paths.append(neosnippet_path)
+        if (ultisnips_path := path / "UltiSnips").exists():
+            ultisnips_paths.append(ultisnips_path)
+        if (ultisnips_path := path / "snippets").exists():
```

Comment:
haha i like the `:=`, 👍 

I was super excited to use them but never really found a big use for it. gonna steal this from u

Reply:
Yeah the walrus op makes sense sometimes.

I guess this is as good a place to say the following as any other: this codebase is one of the best demonstrations I've seen of how helpful type annotations can be in Python code, while retaining the dynamic nature of the language where it makes sense (ie, the `Settings`, `AutoCMD`, `RPC` classes). Hats off!

Reply:
haha thanks!

I honestly don't think the code here is super good or anything, but It's probably better than average :D